TestFoundation/TestNSBundle.swift

```diff
@@ -0,0 +1,97 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2015 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+
+
+
+#if DEPLOYMENT_RUNTIME_OBJC || os(Linux)
+    import Foundation
+    import XCTest
+#else
+    import SwiftFoundation
+    import SwiftXCTest
+#endif
+
+
+
+class TestNSBundle : XCTestCase {
+    
+    var allTests : [(String, () -> ())] {
+        return [
+            ("test_paths", test_paths),
+            ("test_resources", test_resources),
+            ("test_infoPlist", test_infoPlist),
+            ("test_localizations", test_localizations),
+        ]
+    }
+    
+    func test_paths() {
+        let bundle = NSBundle.mainBundle()
+        
+        // bundlePath
+        XCTAssert(!bundle.bundlePath.isEmpty)
+        XCTAssertEqual(bundle.bundleURL.path, bundle.bundlePath)
+        let path = bundle.bundlePath
+        
+        // etc
+        #if os(OSX)
+        XCTAssertEqual("\(path)/Contents/Resources", bundle.resourcePath)
+        XCTAssertEqual("\(path)/Contents/MacOS/TestFoundation", bundle.executablePath)
+        XCTAssertEqual("\(path)/Contents/Frameworks", bundle.privateFrameworksPath)
+        XCTAssertEqual("\(path)/Contents/SharedFrameworks", bundle.sharedFrameworksPath)
+        XCTAssertEqual("\(path)/Contents/SharedSupport", bundle.sharedSupportPath)
+        #endif
+        
+        XCTAssertNil(bundle.pathForAuxiliaryExecutable("no_such_file"))
```

Comment:
Right, exeDirURL is only set for Windows and Apple platforms. At the very least, CFRelease could be wrapped in `if (exeDirURL != NULL)` — but I don't know what the real solution is. A lot of bundle stuff is just undefined on CoreFoundation side for Linux at the moment.

I could also wrap this test temporarily in `#if os(OSX)`, but I don't know how many other tests will fail until there's proper Linux support done on CF side…